hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -250,29 +368,53 @@ public void renameTable(TableIdentifier from, TableIdentifier originalTo) {
             return null;
           });
 
-      LOG.info("Renamed table from {}, to {}", from, to);
+      LOG.info("Renamed {} from {}, to {}", contentType.name(), fromIdentifier, to);
 
     } catch (NoSuchObjectException e) {
-      throw new NoSuchTableException("Table does not exist: %s", from);
+      switch (contentType) {
+        case TABLE:
+          throw new NoSuchTableException(
+              "Cannot rename %s to %s. Table does not exist", fromIdentifier, to);
+        case VIEW:
+          throw new NoSuchViewException(
+              "Cannot rename %s to %s. View does not exist", fromIdentifier, to);
+      }
 
     } catch (InvalidOperationException e) {
       if (e.getMessage() != null
           && e.getMessage().contains(String.format("new table %s already exists", to))) {
-        throw new org.apache.iceberg.exceptions.AlreadyExistsException(
-            "Table already exists: %s", to);
+        throwErrorForExistedToContent(fromIdentifier, removeCatalogName(toIdentifier));
       } else {
-        throw new RuntimeException("Failed to rename " + from + " to " + to, e);
+        throw new RuntimeException("Failed to rename " + fromIdentifier + " to " + to, e);
       }
 
     } catch (TException e) {
-      throw new RuntimeException("Failed to rename " + from + " to " + to, e);
+      throw new RuntimeException("Failed to rename " + fromIdentifier + " to " + to, e);
 
     } catch (InterruptedException e) {
       Thread.currentThread().interrupt();
       throw new RuntimeException("Interrupted in call to rename", e);
     }
   }
 
+  private void throwErrorForExistedToContent(TableIdentifier from, TableIdentifier to) {
+    String toDatabase = to.namespace().level(0);
+    try {
+      Table table = clients.run(client -> client.getTable(toDatabase, to.name()));
```

Comment:
Why do we have to fetch the table again to throw an errro?

Reply:
`MetastoreUtil.alterTable(client, fromDatabase, fromName, table)` doesn't throw error with exact content type. If alter operation fails it throw RTE with message as `"new table %s already exists"` . 

We need error message with exact content type. 
Also for toIdentifier we are fetching only once.

Reply:
@nk1506 @nastra Do you guys think its really necessary?  Rather than just changing the message to 'table or view already exists'

Reply:
We need the message content to run following tests from ViewCatalogTests.
1. renameViewTargetAlreadyExistsAsView
2. renameViewTargetAlreadyExistsAsTable
3. renameTableTargetAlreadyExistsAsView

Instead of altering the error message , is it a good idea to add the checks as at line [350](https://github.com/apache/iceberg/pull/9852/files#diff-3fdd21d335b598f5354de287be6a8830d78e381c89274f0d6bf85c4c9966b8f0R350)
```
if (tableExists(to)) {
      throw new org.apache.iceberg.exceptions.AlreadyExistsException("Cannot rename %s to %s. Table already exists", from, to);
    }

    if (viewExists(to)) {
      throw new org.apache.iceberg.exceptions.AlreadyExistsException("Cannot rename %s to %s. View already exists", from, to);
    }
```

we do have similar strategy for other catalogs.  Cost of this will be in every rename scenario it will make two hive calls. 
But keeping with exception, these calls will be only made in error scenarios. 
Please share your thoughts.

Reply:
imho, it is too much, I see now we have ViewAwareTableBuilder, TableAwareViewBuilder classes (not to mention extra call) just for the correct error message.

Reply:
We do have similar strategy for other catalogs. Otherwise we will have to override all the corresponding tests for Hive. 
What do you propose ?

Reply:
@szehon-ho / @nastra , Should we do few of these clean ups which are not directly related to view changes with the subsequent tasks? IMO, we should not hold this Hive-View changes just because of some legacy limitations. 
WDYT ?

Reply:
Guys im out for a bit , just in my opinion the code is too complex here.

I am not sure why we require the exact same behavior across catalogs.  My preference is to change the error message to 'view or table exist' and just override or have our own tests.

Else, suggestions is to do a tableExists() or viewExists() check just as JDBCViewCatalog, in the common method.  Though to me its a bit silly.